packages/teams-js/src/private/messageChannels.ts

```diff
@@ -0,0 +1,51 @@
+import { requestPortFromParent } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { runtime } from '../public/runtime';
+
+let telemetryPort: MessagePort | undefined;
```

Comment:
We don't want to have more than one channel open for a particular iFrame, so we should cache the received port in case we are asked for it again.

Is there better approach for this?

Reply:
I think the only change I'd ask for is to move this into the messageChannels namespace so that telemetryPort isn't in the global namespace. That could cause some weird problems down the line.

I don't remember all the details of your feature right now (sorry) but since you said something about this being for iframes we should also make sure that whatever you're doing here will still make sense in cases where the app isn't hosted inside of an iframe (mobile platforms for example)

Reply:
This feature is currently intended for the Teams desktop/web code to allow delegation of the telemetry from the iFrame to the worker thread. 

Based on the documentation I've found, if there is a similar need for Android, we would be able to fulfil this API using the [WebMessagePort APIs](https://developer.android.com/reference/android/webkit/WebMessagePort) in Android.

There doesn't look to be an equivalent API for iOS however.

We could potentially mock something on the teams-js SDK side if needed, but unless we believe we need to improve 1P telemetry performance for the mobile apps, I would suggest we just fail to return a port on those hosts, and fallback to the frame doing its own telemetry?